UndertaleModLib/Models/UndertaleTexturePageItem.cs

```diff
@@ -146,24 +147,21 @@ public void Dispose()
     /// </summary>
     /// <param name="replaceImage">The new image that shall be applied to this texture page item.</param>
     /// <param name="disposeImage">Whether to dispose <paramref name="replaceImage"/> afterwards.</param>
-    public void ReplaceTexture(Image replaceImage, bool disposeImage = true)
+    public void ReplaceTexture(MagickImage replaceImage, bool disposeImage = true)
     {
-        Image finalImage = TextureWorker.ResizeImage(replaceImage, SourceWidth, SourceHeight);
+        // Resize image to bounds on texture page
+        IMagickImage<byte> finalImage = TextureWorker.ResizeImage(replaceImage, SourceWidth, SourceHeight);
 
         // Apply the image to the TexturePage.
         lock (TexturePage.TextureData)
         {
-            TextureWorker worker = new TextureWorker();
-            Bitmap embImage = worker.GetEmbeddedTexture(TexturePage); // Use SetPixel if needed.
+            using TextureWorker worker = new();
+            MagickImage embImage = worker.GetEmbeddedTexture(TexturePage);
 
-            Graphics g = Graphics.FromImage(embImage);
-            g.CompositingMode = System.Drawing.Drawing2D.CompositingMode.SourceCopy;
-            g.DrawImage(finalImage, SourceX, SourceY);
-            g.Dispose();
+            embImage.Composite(finalImage, SourceX, SourceY, CompositeOperator.Copy);
 
-            TexturePage.TextureData.TextureBlob = TextureWorker.GetImageBytes(embImage);
-
-            worker.Cleanup();
+            // Replace original texture with the new version, in the original texture format
+            TexturePage.TextureData.Image = GMImage.FromMagickImage(embImage).ConvertToFormat(TexturePage.TextureData.Image.Format);
```

Comment:
I would split this line like this:
```suggestion
TexturePage.TextureData.Image = GMImage.FromMagickImage(embImage)
									   .ConvertToFormat(TexturePage.TextureData.Image.Format);
```

Reply:
![image](https://github.com/user-attachments/assets/832470eb-eecd-43ea-8664-9860ed440ec2)